### PR TITLE
Support for the fill strategy, which will preserve an aspect ratio and add empty space around the image

### PR DESCRIPTION
**Pull request by tmannherz**

### CODE REVIEW

File: tests/LoadsImages.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Loads images trait
+ */
+trait LoadsImages
+{
+    protected function resourceDir(string $sub = ''): string
+    {
+        $dir = dirname(__FILE__) . DIRECTORY_SEPARATOR . 'resources' . DIRECTORY_SEPARATOR;
+        if ($sub) {
+            $dir .= $sub . DIRECTORY_SEPARATOR;
```
Comment (jgrim) [inline tests/LoadsImages.php]: Is it possible for the `$sub` variable to already end with the `DIRECTORY_SEPARATOR`? Might be good to have a check to just do something like this:
```suggestion
            $dir .=  rtrim($sub, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR;
```

File: src/Adapter/Gumlet/ImageResize.php [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+     * Overridden here to allow for a custom bg color
+     *
+     * @param string $filename
+     * @param string $image_type
+     * @param integer $quality
+     * @param integer $permissions
+     * @param boolean $exact_size
+     * @return $this
+     */
+    public function save($filename, $image_type = null, $quality = null, $permissions = null, $exact_size = false)
```
Comment (jgrim) [inline src/Adapter/Gumlet/ImageResize.php:41]: This is a massive method, might be nice to break this into private or protected methods as cleanup 🤷‍♂️

**Review by jgrim — DISMISSED** [review]
Approved with comments

File: src/Adapter/Gumlet/ImageResize.php (hunk 2 above)
Comment (tmannherz) [reply src/Adapter/Gumlet/ImageResize.php:41]: It's almost directly copied and overridden from the parent. I just changed their hard-coded `255, 255, 255` bg color to one set on the object.

**Review by jgrim — APPROVED** [review]

**Pull request merged**